### PR TITLE
feat(generator): trim overly long field comments

### PR DESCRIPTION
**Pull request by coryan**
The documentation in some fields requires special hacks to get Doxygen to parse them more or less correctly. For the next service I am planning to generate (`aiplatform`) this was getting out of hand. I think just preserving the first paragraph and linking the protobuf message documentation is enough, so that is what I did.

Motivated by #11810

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11859)

### CODE REVIEW

File: google/cloud/asset/v1/asset_client.h [hunk 1]
```diff
@@ -579,54 +579,8 @@ class AssetServiceClient {
...
-  ///  * `Important` to find Google Cloud resources that contain "Important" as a
-  ///    word in any of the searchable fields.
-  ///  * `Impor*` to find Google Cloud resources that contain "Impor" as a prefix
-  ///    of any word in any of the searchable fields.
-  ///  * `Important location:(us-west1 OR global)` to find Google Cloud
-  ///    resources that contain "Important" as a word in any of the searchable
-  ///    fields and are also located in the "us-west1" region or the "global"
-  ///    location.
+  ///     @n
+  ///     For more information, see [SearchAllResourcesRequest][google.cloud.asset.v1.SearchAllResourcesRequest].
```
Comment (devbww) [inline google/cloud/asset/v1/asset_client.h:583]: It looks like there is some extra indentation added in these cases.

File: generator/internal/descriptor_utils.cc [hunk 2]
```diff
@@ -409,25 +420,58 @@ ParameterCommentSubstitution substitutions[] = {
     {"\n", "\n  /// "},
 };
 
+// Very long parameters need different formatting.
+auto constexpr kShortParamFormat = "  /// @param %s %s\n";
+auto constexpr kLongParamFormat = R"""(  /// @param %s %s
+  ///     @n
+  ///     For more information, see [%s][%s].
```
Comment (devbww) [inline generator/internal/descriptor_utils.cc:427]: [See comment in generated code about indentation depth.]

File: generator/internal/descriptor_utils.cc [hunk 3]
```diff
@@ -397,6 +387,27 @@ ParameterCommentSubstitution substitutions[] = {
     {R"""(`projects/<Project ID or '-'>`)""",
      R"""(``projects/<Project ID or '-'>``)"""},
 
+    // Further trim some initial paragraphs for long descriptions.
+    {R"""( The included patch
+ environment would specify the scikit-learn version as follows:)""",
+     ""},
+
```
Comment (devbww) [inline generator/internal/descriptor_utils.cc:394]: Is there reason to separate these substitutions, and, if so, do they deserve their own comment?

**Review by devbww — APPROVED** [review]

File: google/cloud/asset/v1/asset_client.h (hunk 1 above)
Comment (coryan) [reply google/cloud/asset/v1/asset_client.h]: Fixed

File: generator/internal/descriptor_utils.cc (hunk 2 above)
Comment (coryan) [reply generator/internal/descriptor_utils.cc]: Regenerated

File: generator/internal/descriptor_utils.cc (hunk 3 above)
Comment (coryan) [reply generator/internal/descriptor_utils.cc]: Nope. Fixed.

**Pull request merged**